alf/environments/suite_carla.py

```diff
@@ -1379,25 +1420,122 @@ def _spawn(self):
+                walker_bp.set_attribute('is_invincible', 'false')
+            # set the max speed
+            if walker_bp.has_attribute('speed'):
+                if (random.random() > self._percentage_walkers_running):
+                    # walking
+                    walker_speeds.append(
+                        walker_bp.get_attribute('speed').recommended_values[1])
+                else:
+                    # running
+                    walker_speeds.append(
+                        walker_bp.get_attribute('speed').recommended_values[2])
+            else:
+                logging.info("Walker has no speed")
+                walker_speeds.append(0.0)
+            commands.append(carla.command.SpawnActor(walker_bp, spawn_point))
+        responses = self._client.apply_batch_sync(commands, True)
+        walker_speeds2 = []
+        self._walkers = []
+        for response, walker_speed, spawn_point in zip(
+                responses, walker_speeds, spawn_points):
+            if response.error:
+                logging.error(
+                    "%s: %s" % (response.error, spawn_point.location))
+                continue
+            walker = self._world.get_actor(response.actor_id)
+            self._walkers.append({"walker": walker})
+            walker_speeds2.append(walker_speed)
+        walker_speeds = walker_speeds2
 
-        self._observation_spec = self._players[0].observation_spec()
-        self._action_spec = self._players[0].action_spec()
+        # 3. we spawn the walker controller
+        commands = []
+        walker_controller_bp = self._world.get_blueprint_library().find(
+            'controller.ai.walker')
+        for walker in self._walkers:
+            commands.append(
+                carla.command.SpawnActor(walker_controller_bp,
+                                         carla.Transform(),
+                                         walker["walker"].id))
+        responses = self._client.apply_batch_sync(commands, True)
+        for response, walker in zip(responses, self._walkers):
+            if response.error:
+                logging.error(response.error)
+                continue
+            walker["controller"] = self._world.get_actor(response.actor_id)
+
+        # wait for a tick to ensure client receives the last transform of the walkers we have just created
+        self._world.tick()
+
+        # 5. initialize each controller and set target to walk to (list is [controler, actor, controller, actor ...])
+        # set how many pedestrians can cross the road
+        self._world.set_pedestrians_cross_factor(
+            self._percentage_walkers_crossing)
+        for walker, walker_speed in zip(self._walkers, walker_speeds):
+            # start walker
+            walker['controller'].start()
+            # set walk to random point
+            location = self._world.get_random_location_from_navigation()
+            walker['controller'].go_to_location(location)
```

Comment:
I'm curious if after the walker goes to the location, what will it do? Is there any cycle for the walker's movement?